src/subscript/fmu_copy_revision/fmu_copy_revision.py

```diff
@@ -0,0 +1,654 @@
+        target = pathlib.Path(proposal)
+        self.target = str(target.absolute())
+        print(f"Selected target is <{self.target}>")
+
+        if self.args.cleanup:
+            logger.info("Delete existing target if it exists...")
+            if target.is_dir():
+                print(f"Cleanup: Deleting existing target: <{self.target}>")
+                shutil.rmtree(self.target)
+
+    def menu_target_folder(self):
+        """Print an interactive menu to the user for target."""
+
+        self.construct_default_target()
+        dft = self.default_target
+        propose = input(f"Choose output folder (default is <{dft}>: ") or dft
+        self.construct_target(propose)
+
+    def check_rms_lockfile(self):
+        """Check if RMS project has an active lockfile if interactive mode."""
+        lockfiles = pathlib.Path(self.source).glob("rms/model/*/project_lock_file")
+
+        if len(list(lockfiles)) > 0:
+            print(
+                "Warning, it seems that one or more RMS projects have a lock file "
+                "and may perhaps be in a saving process..."
+            )
+            for lockfile in lockfiles:
+                print(f"<{lockfile}> owner of lockfile is {lockfile.owner()}")
+
+            if not self.batch:
+                answer = (
+                    input("Continue anyway? (default is 'Yes' if you press enter): ")
+                    or "Y"
+                )
+                if not answer.startswith(("y", "Y")):
+                    print("Stopped by user")
+                    sys.exit()
+
+            print("Will continue...")
+
+    def check_disk_space(self):
+        """Checking diskspace."""
+        print("Checking disk space at current partition...")
+        total, used, free = shutil.disk_usage(".")
+        print("  Total: %d G" % (total // (2 ** 30)))
+        print("  Used:  %d G" % (used // (2 ** 30)))
+        print("  Free:  %d G" % (free // (2 ** 30)))
+
+        if self.args.skipestimate:
+            print("  Skip estimation of current revision size!")
+            return
+
+        print(f"  Estimate size of current revision <{self.source}> ...")
+
+        freekbytes = free // 1024
+
+        def _get_size(path: str) -> int:
+            sum = 0
+            for p in pathlib.Path(path).rglob("*"):
+                if not p.is_symlink():
+                    sum += p.stat().st_size
+
+            return sum
+
+        def _filesize(size: float) -> str:
+            for unit in ("B", "K", "M", "G"):
+                if size < 1024:
+                    break
+                size /= 1024
+            return f"{size:.1f} {unit}"
+
+        fsize = _get_size(self.source)
+        print(f"\n  Size of existing revision is: {_filesize(fsize)}\n")
+
+        sourcekbytes = fsize // 1024
+        if sourcekbytes > freekbytes:
+            print("Not enough space left for copying! STOP!")
+            sys.exit()
+
+        time.sleep(1)
```

Comment:
Needed?

Reply:
Not really, but the menu comes very fast so that disk space info disappears if a small xterm window. Think I will keep it